File: OOP-C++/Basics/Part4/Inheritence/vehicle_use.cpp
#include <iostream>
using namespace std;
#include "vehicle_class.cpp"
#include "car_class.cpp"

int main(){
	Vehicle v;
	v.color = "Blue";
	//v.maxSpeed = 100;		//Error: maxSpeed' is private within this context
	//v.numTyres = 4;		//Error: numTyres' is protected within this context

	Car c;
	c.color = "Black";
	//c.numTyres = 4;		//numTyres' is protected within this context
	c.numGears = 5;
}

File: OOP-Java/output/com/output16/VehicleUse.java
package com.output16;

public class VehicleUse {

	public static void main(String[] args) {
		Vehicle v = new Vehicle();
		v.color = "Black";
		v.setMaxSpeed(10);
		v.print();

		Car c = new Car();
		c.numGears = 10;
		c.color = "Black";
		c.setMaxSpeed(100);
		c.print();
	}
}

/*
   Vehicle color : Black
   Vehicle Speed : 10
   Vehicle color : Black
   Vehicle Speed : 100
   Car numGears : 10
   Car isConvertible : false
 */
